platforms/software/dependency-management/src/integTest/groovy/org/gradle/plugin/devel/variants/TargetJVMVersionOnLibraryTooNewFailureDescriberIntegrationTest.groovy

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugin.devel.variants
+
+import org.gradle.api.JavaVersion
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.internal.component.resolution.failure.exception.VariantSelectionException
+
+class TargetJVMVersionOnLibraryTooNewFailureDescriberIntegrationTest extends AbstractIntegrationSpec {
+    Integer currentJava = Integer.valueOf(JavaVersion.current().majorVersion)
+    Integer tooHighJava = currentJava + 1
+
+    def 'JVM version too low on non-plugin dependency uses standard error message'() {
+        given:
+        def producer = file('producer')
+        def consumer = file('consumer')
+
+        file('settings.gradle') << """
+            include 'producer', 'consumer'
+        """
+
+        producer.file('build.gradle') << """
+            plugins {
+                id('java-library')
+            }
+
+            configurations.configureEach {
+                if (canBeConsumed)  {
+                    attributes {
+                        attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, $tooHighJava)
```

Comment:
❌ I think this should be a more realistic scenario.  We just need to set either the source/target compatibility or toolchain to two different versions.